python/pyspark/sql/types.py

```diff
@@ -1299,6 +1300,12 @@ def fromInternal(self, obj: Tuple) -> "Row":
         return _create_row(self.names, values)
 
 
+class VariantType(AtomicType):
```

Comment:
I'm wondering we should make this `AtomicType`? should be just `DataType`?

Reply:
(my bad, I thought I'd replied to this last week, but now I see that my comment didn't go through)

In this case I was mirroring the definition of the datatype on the scala side. We can change both in a future PR